### PR TITLE
rafhttp: refactor peer.go

### PR DESCRIPTION
**Pull request by xiang90**
refactoring

### CODE REVIEW

File: rafthttp/peer.go [hunk 1]
```diff
@@ -31,6 +31,18 @@ const (
 	ConnWriteTimeout = 5 * time.Second
 
 	recvBufSize = 4096
+
+	streamApp   = "streamMsgApp"
+	streamMsg   = "streamMsg"
+	pipelineMsg = "pipeline"
```
Comment (yichengq) [inline rafthttp/peer.go]: we have streamType defined: https://github.com/coreos/etcd/blob/master/rafthttp/stream.go#L37
Any thoughts to integrate them?

Comment (xiang90) [inline rafthttp/peer.go]: the next step is to merge them (there are also something i want to refactor in stream.go). but not in this pr.

**Comment by yichengq** [comment]
Could we use a better commit title? maybe `rafthttp: refactor peer.pick function

**Comment by xiang90** [comment]
@yichengq sure. done.

**Comment by yichengq** [comment]
lgtm

**Pull request merged**